internal/p10k.zsh

```diff
@@ -4280,24 +4281,56 @@ function instant_prompt_chezmoi_shell() {
 # Virtualenv: current working virtualenv
 # More information on virtualenv (Python):
 # https://virtualenv.pypa.io/en/latest/
+
+_virtualenv_VIRTUAL_ENV() {
+  local n
+  # VIRTUAL_ENV is the on-disk path, VIRTUAL_ENV_PROMPT is `(name_of_virtualenv) `.
+  # Use custom name if it was set (python -m venv -p "foo" .venv)
+  if [[ $VIRTUAL_ENV_PROMPT == '('?*') ' && $VIRTUAL_ENV_PROMPT != "($n) " ]]; then
+    n=$VIRTUAL_ENV_PROMPT[2,-3]
+  # Use parent directory name if virtualenv name is generic (eg .venv)
+  elif [[ $v == $~_POWERLEVEL9K_VIRTUALENV_GENERIC_NAMES ]]; then
+    n=${VIRTUAL_ENV:h:t}
+  # Otherwise use the virtualenv name as-is
+  else
+    n=${VIRTUAL_ENV:t}
+  fi
+  echo "$n"
+}
+_virtualenv_poetry() {
+  local idx=$1
+  local dir=${_p9k__parent_dirs[idx]}
+  local pyproject="$dir/pyproject.toml"
+  _p9k_cached_cmd 0 '' poetry -C "$dir" version
```

Comment:
This command isn't cached and so prompt re-rendering is very slow. We only need to re-run this command if `$pyproject` is modified. I tried 

```python
  _p9k_cached_cmd 0 "$pyproject" poetry -C "$dir" version
```

But this didn't seem to help 🤔

Reply:
Fixed! .. somehow